From 14456a86efc9f6483238c4796e036727d85a05bb Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Wed, 16 May 2018 17:14:53 +0800
Subject: [PATCH] =?UTF-8?q?ARM=EF=BC=9Adts:=20rockchip:=20rk322x:=20Config?=
 =?UTF-8?q?=20the=20pwm2=20pinctrl=20with=20pull=20up?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The default state of pwm2 pin is a pull up, we need to keep it
for the second global reset issue, when pwm2 used for the regulator.

Change-Id: Ibe3487904196464114622dbc10ba270e3cf509d4
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-evb-android.dts | 4 ++++
 arch/arm/boot/dts/rk3229-evb.dts         | 2 +-
 arch/arm/boot/dts/rk322x.dtsi            | 4 ++--
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-evb-android.dts b/arch/arm/boot/dts/rk3229-evb-android.dts
index 8f546872f337..6ab0a5ea4cf4 100644
--- a/arch/arm/boot/dts/rk3229-evb-android.dts
+++ b/arch/arm/boot/dts/rk3229-evb-android.dts
@@ -341,10 +341,14 @@
 
 &pwm1 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1_pin_pull_down>;
 };
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_up>;
 };
 
 &pwm3 {
diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index 096840d116b6..fc4f19fb2563 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -334,7 +334,7 @@
 &pwm2 {
 	status = "okay";
 	pinctrl-names = "active";
-	pinctrl-0 = <&pwm2_pin_pull_down>;
+	pinctrl-0 = <&pwm2_pin_pull_up>;
 };
 
 &sdio {
diff --git a/arch/arm/boot/dts/rk322x.dtsi b/arch/arm/boot/dts/rk322x.dtsi
index e9db63bb97db..c88af702e904 100644
--- a/arch/arm/boot/dts/rk322x.dtsi
+++ b/arch/arm/boot/dts/rk322x.dtsi
@@ -1580,8 +1580,8 @@
 				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_none>;
 			};
 
-			pwm2_pin_pull_down: pwm2-pin-pull-down {
-				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_down>;
+			pwm2_pin_pull_up: pwm2-pin-pull-up {
+				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_up>;
 			};
 		};
 
-- 
2.35.3

